### ISSUE TITLE
Run '--workload CassandraKeyValue' case will throw java.lang.IllegalArgumentException: bound must be positive exception.

### ISSUE THREAD

**Issue by huahouye** [yugabyte/yb-sample-apps#24]
I have try to run the sample from https://docs.yugabyte.com/latest/explore/auto-sharding/macos/#4-insert-and-query-a-table doc page with yugabytedb v2.1.2.0. It seems out of date.
```
java -jar ./yb-sample-apps.jar \
  --workload CassandraKeyValue \
  --nodes 127.0.0.1:9042 \
  --nouuid \
  --num_unique_keys 1 \
  --num_writes 2 \
  --num_threads_read 0 \
  --num_threads_write 1 \
  --value_size 10000000
```
run this sample case it will throw exception as below:
```
776 [main] INFO com.yugabyte.sample.apps.AppBase  - Created a Cassandra table using query: [CREATE TABLE IF NOT EXISTS CassandraKeyValue (k varchar, v blob, primary key (k));]
876 [Thread-2] INFO com.yugabyte.sample.apps.AppBase  - Caught Exception:
java.lang.IllegalArgumentException: bound must be positive
	at java.util.concurrent.ThreadLocalRandom.nextLong(ThreadLocalRandom.java:402)
	at com.yugabyte.sample.common.SimpleLoadGenerator.getKeyToWrite(SimpleLoadGenerator.java:161)
	at com.yugabyte.sample.apps.CassandraKeyValueBase.doWrite(CassandraKeyValueBase.java:147)
	at com.yugabyte.sample.apps.AppBase.performWrite(AppBase.java:701)
	at com.yugabyte.sample.common.IOPSThread.run(IOPSThread.java:87)
5790 [Thread-1] INFO com.yugabyte.sample.common.metrics.MetricsTracker  - Read: 0.00 ops/sec (0.00 ms/op), 0 total ops  |  Write: 0.20 ops/sec (91.00 ms/op), 1 total ops  |  Uptime: 5011 ms | maxWrittenKey: 0 | maxGeneratedKey: 0 |
```

I use `--num_unique_keys 1 2` to walk around at this point. I may read 'yb-sample-apps' code later to figure out what this issue exactly is.

```
java -jar ./yb-sample-apps.jar \
  --workload CassandraKeyValue \
  --nodes 127.0.0.1:9042 \
  --nouuid \
  --num_unique_keys 1 2 \
  --num_writes 2 \
  --num_threads_read 0 \
  --num_threads_write 1 \
  --value_size 10000000
```

**Comment by ddorian** [comment]
@huahouye 
I don't think that app handles `--nouuid` option based on it's help:
```
java -jar ./yb-sample-apps.jar --help CassandraKeyValue
0 [main] INFO com.yugabyte.sample.Main  - Starting sample app...
Usage and options for workload CassandraKeyValue in YugaByte DB Sample Apps.

 - CassandraKeyValue :
   -----------------
		Sample key-value app built on Cassandra with concurrent reader and writer threads.
		 Each of these threads operates on a single key-value pair. The number of readers 
		 and writers, the value size, the number of inserts vs updates are configurable.

		Usage:
			java -jar yb-sample-apps.jar \
			--workload CassandraKeyValue \
			--nodes 127.0.0.1:9042

		Other options (with default values):
			[ --num_unique_keys 1000000 ]
			[ --num_reads -1 ]
			[ --num_writes -1 ]
			[ --value_size 0 ]
			[ --num_threads_read 24 ]
			[ --num_threads_write 2 ]
			[ --table_ttl_seconds -1 ]
```

I think the app is using shared code/functions that take --nouuid but doesn't handle it and it errors. For me it works when removing `--nouuid`.

**Comment by huahouye** [comment]
Not works 100%, sometime it works but some time does not, log as below. I will give it another try at this weekend. It is just a test, so issue is closed now.
```
huahouye-MacBook-Pro:yugabyte-2.1.2.0 houyehua$ java -jar yb-sample-apps.jar \
>   --workload CassandraKeyValue \
>   --nodes 127.0.0.1:9042 \
>   --num_unique_keys 1 \
>   --num_reads -1 \
>   --num_writes 2 \
>   --num_threads_read 0 \
>   --num_threads_write 1 \
>   --table_ttl_seconds -1 \
>   --value_size 10000000
0 [main] INFO com.yugabyte.sample.Main  - Starting sample app...
58 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Using a randomly generated UUID : 747a5a7c-7177-45ae-aff2-d5eb1f96ed81
75 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - App: CassandraKeyValue
75 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Run time (seconds): -1
75 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Adding node: 127.0.0.1:9042
78 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num reader threads: 0, num writer threads: 1
79 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num unique keys to insert: 1
79 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num keys to update: 1
79 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num keys to read: -1
80 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Value size: 10000000
80 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Restrict values to ASCII strings: false
80 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Perform sanity check at end of app run: false
80 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Table TTL (secs): -1
80 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Local reads: false
80 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Read only load: false
512 [main] INFO com.yugabyte.sample.apps.AppBase  - Connecting with 4 clients to nodes: /127.0.0.1:9042
1610 [main] INFO com.yugabyte.sample.apps.AppBase  - Created a Cassandra table using query: [CREATE TABLE IF NOT EXISTS CassandraKeyValue (k varchar, v blob, primary key (k));]
4172 [main] INFO com.yugabyte.sample.Main  - The sample app has finished
huahouye-MacBook-Pro:yugabyte-2.1.2.0 houyehua$ java -jar yb-sample-apps.jar   --workload CassandraKeyValue   --nodes 127.0.0.1:9042   --num_unique_keys 1   --num_reads -1   --num_writes 2   --num_threads_read 0   --num_threads_write 1   --table_ttl_seconds -1   --value_size 10000000
0 [main] INFO com.yugabyte.sample.Main  - Starting sample app...
44 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Using a randomly generated UUID : d89531eb-e7fc-4930-bab2-728b34d8b449
50 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - App: CassandraKeyValue
50 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Run time (seconds): -1
50 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Adding node: 127.0.0.1:9042
52 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num reader threads: 0, num writer threads: 1
52 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num unique keys to insert: 1
52 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num keys to update: 1
53 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num keys to read: -1
53 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Value size: 10000000
53 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Restrict values to ASCII strings: false
53 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Perform sanity check at end of app run: false
53 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Table TTL (secs): -1
53 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Local reads: false
53 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Read only load: false
186 [main] INFO com.yugabyte.sample.apps.AppBase  - Connecting with 4 clients to nodes: /127.0.0.1:9042
740 [main] INFO com.yugabyte.sample.apps.AppBase  - Created a Cassandra table using query: [CREATE TABLE IF NOT EXISTS CassandraKeyValue (k varchar, v blob, primary key (k));]
3147 [main] INFO com.yugabyte.sample.Main  - The sample app has finished
huahouye-MacBook-Pro:yugabyte-2.1.2.0 houyehua$ java -jar yb-sample-apps.jar   --workload CassandraKeyValue   --nodes 127.0.0.1:9042   --num_unique_keys 1   --num_reads -1   --num_writes 2   --num_threads_read 0   --num_threads_write 1   --table_ttl_seconds -1   --value_size 10000000
0 [main] INFO com.yugabyte.sample.Main  - Starting sample app...
39 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Using a randomly generated UUID : 7144cca2-4f63-4231-8a7e-999fe8ce5890
44 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - App: CassandraKeyValue
44 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Run time (seconds): -1
44 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Adding node: 127.0.0.1:9042
47 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num reader threads: 0, num writer threads: 1
47 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num unique keys to insert: 1
48 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num keys to update: 1
48 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num keys to read: -1
48 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Value size: 10000000
48 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Restrict values to ASCII strings: false
48 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Perform sanity check at end of app run: false
48 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Table TTL (secs): -1
48 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Local reads: false
48 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Read only load: false
216 [main] INFO com.yugabyte.sample.apps.AppBase  - Connecting with 4 clients to nodes: /127.0.0.1:9042
787 [main] INFO com.yugabyte.sample.apps.AppBase  - Created a Cassandra table using query: [CREATE TABLE IF NOT EXISTS CassandraKeyValue (k varchar, v blob, primary key (k));]
915 [Thread-2] INFO com.yugabyte.sample.apps.AppBase  - Caught Exception:
java.lang.IllegalArgumentException: bound must be positive
	at java.util.concurrent.ThreadLocalRandom.nextLong(ThreadLocalRandom.java:402)
	at com.yugabyte.sample.common.SimpleLoadGenerator.getKeyToWrite(SimpleLoadGenerator.java:161)
	at com.yugabyte.sample.apps.CassandraKeyValueBase.doWrite(CassandraKeyValueBase.java:147)
	at com.yugabyte.sample.apps.AppBase.performWrite(AppBase.java:701)
	at com.yugabyte.sample.common.IOPSThread.run(IOPSThread.java:87)
5797 [Thread-1] INFO com.yugabyte.sample.common.metrics.MetricsTracker  - Read: 0.00 ops/sec (0.00 ms/op), 0 total ops  |  Write: 0.20 ops/sec (123.00 ms/op), 1 total ops  |  Uptime: 5008 ms | maxWrittenKey: 0 | maxGeneratedKey: 0 |
^Chuahouye-MacBook-Pro:yugabyte-2.1.2.0 houyehua$
huahouye-MacBook-Pro:yugabyte-2.1.2.0 houyehua$
huahouye-MacBook-Pro:yugabyte-2.1.2.0 houyehua$
huahouye-MacBook-Pro:yugabyte-2.1.2.0 houyehua$
huahouye-MacBook-Pro:yugabyte-2.1.2.0 houyehua$ java -jar yb-sample-apps.jar   --workload CassandraKeyValue   --nodes 127.0.0.1:9042   --num_unique_keys 1   --num_reads -1   --num_writes 2   --num_threads_read 0   --num_threads_write 1   --table_ttl_seconds -1   --value_size 10000000
1 [main] INFO com.yugabyte.sample.Main  - Starting sample app...
57 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Using a randomly generated UUID : fdb02e3d-0e68-48a5-9357-ff64bd5bfff3
70 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - App: CassandraKeyValue
70 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Run time (seconds): -1
70 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Adding node: 127.0.0.1:9042
72 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num reader threads: 0, num writer threads: 1
73 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num unique keys to insert: 1
73 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num keys to update: 1
73 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num keys to read: -1
73 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Value size: 10000000
73 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Restrict values to ASCII strings: false
73 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Perform sanity check at end of app run: false
73 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Table TTL (secs): -1
73 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Local reads: false
73 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Read only load: false
255 [main] INFO com.yugabyte.sample.apps.AppBase  - Connecting with 4 clients to nodes: /127.0.0.1:9042
1144 [main] INFO com.yugabyte.sample.apps.AppBase  - Created a Cassandra table using query: [CREATE TABLE IF NOT EXISTS CassandraKeyValue (k varchar, v blob, primary key (k));]
1318 [Thread-2] INFO com.yugabyte.sample.apps.AppBase  - Caught Exception:
java.lang.IllegalArgumentException: bound must be positive
	at java.util.concurrent.ThreadLocalRandom.nextLong(ThreadLocalRandom.java:402)
	at com.yugabyte.sample.common.SimpleLoadGenerator.getKeyToWrite(SimpleLoadGenerator.java:161)
	at com.yugabyte.sample.apps.CassandraKeyValueBase.doWrite(CassandraKeyValueBase.java:147)
	at com.yugabyte.sample.apps.AppBase.performWrite(AppBase.java:701)
	at com.yugabyte.sample.common.IOPSThread.run(IOPSThread.java:87)
6158 [Thread-1] INFO com.yugabyte.sample.common.metrics.MetricsTracker  - Read: 0.00 ops/sec (0.00 ms/op), 0 total ops  |  Write: 0.20 ops/sec (165.00 ms/op), 1 total ops  |  Uptime: 5011 ms | maxWrittenKey: 0 | maxGeneratedKey: 0 |
^Chuahouye-MacBook-Pro:yugabyte-2.1.2.0 houyehua$
```

**Comment by ddorian** [comment]
@huahouye 
Did you try with the latest version https://github.com/yugabyte/yb-sample-apps/releases ? Both cases worked for me:

```
java -jar yb-sample-apps.jar   --workload CassandraKeyValue   --nodes 127.0.0.1:9042   --num_unique_keys 1   --num_reads -1   --num_writes 2   --num_threads_read 0   --num_threads_write 1   --table_ttl_seconds -1   --value_size 10000000
0 [main] INFO com.yugabyte.sample.Main  - Starting sample app...
95 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Using a randomly generated UUID : f066d818-6506-49b4-8cea-cd438b77e0ec
106 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - App: CassandraKeyValue
106 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Run time (seconds): -1
106 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Adding node: 127.0.0.1:9042
108 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num reader threads: 0, num writer threads: 1
108 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num unique keys to insert: 1
108 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num keys to update: 1
108 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num keys to read: -1
108 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Value size: 10000000
108 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Restrict values to ASCII strings: false
108 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Perform sanity check at end of app run: false
108 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Table TTL (secs): -1
108 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Local reads: false
108 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Read only load: false
191 [main] INFO com.yugabyte.sample.apps.AppBase  - Connecting with 4 clients to nodes: /127.0.0.1:9042
708 [main] INFO com.yugabyte.sample.apps.AppBase  - Created a Cassandra table using query: [CREATE TABLE IF NOT EXISTS CassandraKeyValue (k varchar, v blob, primary key (k));]
3081 [main] INFO com.yugabyte.sample.Main  - The sample app has finished



java -jar yb-sample-apps.jar   --workload CassandraKeyValue   --nodes 127.0.0.1:9042   --num_unique_keys 1   --num_reads -1   --num_writes 2   --num_threads_read 0   --num_threads_write 1   --table_ttl_seconds -1   --value_size 10000000
0 [main] INFO com.yugabyte.sample.Main  - Starting sample app...
24 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Using a randomly generated UUID : 422f74c2-426a-4244-9d9a-79096e9cdf14
28 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - App: CassandraKeyValue
28 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Run time (seconds): -1
28 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Adding node: 127.0.0.1:9042
29 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num reader threads: 0, num writer threads: 1
29 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num unique keys to insert: 1
29 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num keys to update: 1
29 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Num keys to read: -1
29 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Value size: 10000000
29 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Restrict values to ASCII strings: false
29 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Perform sanity check at end of app run: false
29 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Table TTL (secs): -1
29 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Local reads: false
29 [main] INFO com.yugabyte.sample.common.CmdLineOpts  - Read only load: false
82 [main] INFO com.yugabyte.sample.apps.AppBase  - Connecting with 4 clients to nodes: /127.0.0.1:9042
452 [main] INFO com.yugabyte.sample.apps.AppBase  - Created a Cassandra table using query: [CREATE TABLE IF NOT EXISTS CassandraKeyValue (k varchar, v blob, primary key (k));]
2923 [main] INFO com.yugabyte.sample.Main  - The sample app has finished

```

**Comment by huahouye** [comment]
No, I didn't think I used the right version when I tried to create a Chinese website about yugabyte db last week (https://www.yugabytedb.cn/ , https://github.com/yugabytedb/yugabytedb.github.io). As I asked the question here https://yugabyte-db.slack.com/archives/CG0KQF0GG/p1584978729482400, and later I found many documents on https://docs.yugabyte.com/latest website were out of date. So now I upgrade all softwares to latest version, but haven't try it again yet. 

On this weekend I will update yugabytedb.cn pages if I have time, and if I run into this issue again, I will reopen it or create a new one. 

Thank you for your help.
